### PR TITLE
Make the soca run ctest work again

### LINKED ISSUE #184: [Bug] PR 145 broke this repo
I should have paid attention! @jiaruidong2017 , your changes below L180: 
https://github.com/NOAA-EMC/GDASApp/blob/2712a22aa81788d85e14cac358de12956a75d238/ush/ufsda/yamltools.py#L181
broke this repo.

It would have been avoidable had we triggered the `ci`, the `soca_run` test is broken again.

@jiaruidong2017 : My `concern` is that I can easily `fix` what you broke, but I have no idea how to not break it for you, since you/we did not add a ctest that covers your code changes. I'll do my "best" to try to maintain the functionality you added, but at the end of the day, the burden of fixing that bug is on you. 

This could also have been avoided in my last PR, which was merged after #145 , but I did not contain the latest commit and missed that bug as well ... I just added a github reminder that will warn us when the branch that we want to merge is behind develop.

### PR DESCRIPTION
**Pull request by guillaumevernieres**
- fixes #184 

@RussTreadon-NOAA , @CoryMartin-NOAA , what are the chances that the scripts in the g-w are broken? Were they tested after  @jiaruidong2017 's PR? ... mental testing doesn't count :)

Feel free to suggest a different implementation ... I just made stuff work with minimal thinking/work.

### CODE REVIEW

**Comment by jiaruidong2017** [comment]
Thanks @guillaumevernieres for your efforts. I forgot to consider the cases with both atm DA and land DA. However, the current change is not working for the land DA. I made further changes as below: 

```
    if 'assim_freq' in os.environ:
        config['ATM_WINDOW_LENGTH'] = f"PT{os.environ['assim_freq']}H"
    elif 'atm_window_length' in config.keys():
        config['ATM_WINDOW_LENGTH'] = config['atm_window_length']
    else:
        raise KeyError("Neither $(atm_window_length) nor ${assim_freq} defined")
    # get the default window begin (atm window begin)
    h = re.findall('PT(\\d+)H', config['ATM_WINDOW_LENGTH'])[0]
    win_begin = valid_time_obj - datetime.timedelta(hours=int(h)/2)
    win_end = valid_time_obj + datetime.timedelta(hours=int(h)/2)
    config['ATM_WINDOW_BEGIN'] = win_begin.strftime('%Y-%m-%dT%H:%M:%SZ')
    config['ATM_WINDOW_END'] = win_end.strftime('%Y-%m-%dT%H:%M:%SZ')
    config['ATM_BEGIN_YYYYmmddHHMMSS'] = win_begin.strftime('%Y%m%d.%H%M%S')
    # get land window begin
    if 'land_window_length' in config.keys():
        config['LAND_WINDOW_LENGTH'] = config['land_window_length']
        h = re.findall('PT(\\d+)H', config['LAND_WINDOW_LENGTH'])[0]
        win_begin = valid_time_obj - datetime.timedelta(hours=int(h)/2)
        win_end = valid_time_obj + datetime.timedelta(hours=int(h)/2)
        config['LAND_WINDOW_BEGIN'] = win_begin.strftime('%Y-%m-%dT%H:%M:%SZ')
        config['LAND_WINDOW_END'] = win_end.strftime('%Y-%m-%dT%H:%M:%SZ')
        config['LAND_BEGIN_YYYYmmddHHMMSS'] = win_begin.strftime('%Y%m%d.%H%M%S')
    return config
```
@guillaumevernieres Would you please include my changes above in your PR? Otherwise, I could create a PR, and please let me know. Thanks again.

File: ush/ufsda/yamltools.py [hunk 1]
```diff
@@ -177,16 +177,15 @@ def calc_time_vars(config):
...
+    # get the default window begin (atm window begin)
+    h = re.findall('PT(\\d+)H', config['ATM_WINDOW_LENGTH'])[0]
+    win_begin = valid_time_obj - datetime.timedelta(hours=int(h)/2)
+    win_end = valid_time_obj + datetime.timedelta(hours=int(h)/2)
+    config['ATM_WINDOW_BEGIN'] = win_begin.strftime('%Y-%m-%dT%H:%M:%SZ')
+    config['ATM_WINDOW_END'] = win_end.strftime('%Y-%m-%dT%H:%M:%SZ')
+    config['ATM_BEGIN_YYYYmmddHHMMSS'] = win_begin.strftime('%Y%m%d.%H%M%S')
     # get land window begin
-    elif 'land_window_length' in config.keys():
+    if 'land_window_length' in config.keys():
```
Comment (jiaruidong2017) [inline ush/ufsda/yamltools.py:188]: Add the following below the line 188:
`        config['LAND_WINDOW_LENGTH'] = config['land_window_length']`

File: ush/ufsda/yamltools.py [hunk 2]
```diff
@@ -177,16 +177,15 @@ def calc_time_vars(config):
         config['LAND_WINDOW_LENGTH'] = config['land_window_length']
```
Comment (jiaruidong2017) [inline ush/ufsda/yamltools.py:177]: Remove the following two lines (lines: 176 and 177): 
```
    elif 'land_window_length' in config.keys():
        config['LAND_WINDOW_LENGTH'] = config['land_window_length']
```

**Comment by RussTreadon-NOAA** [comment]
@guillaumevernieres , thanks for the heads up.  I attempted to rerun atmospheric 3dvar using `fv3jedi_var.x` from g-w branch `develop` using GDASApp branch`develop`.   The executable failed with
```
0: OOPS_STATS Variational start                        - Runtime:     1.52 sec,  Local Memory:   189.12 Mb
1: Exception: BadValue: Badly formatted date: $(AT
2: Exception: BadValue: Badly formatted date: $(AT
3: Exception: BadValue: Badly formatted date: $(AT
5: Exception: BadValue: Badly formatted date: $(AT
4: Exception: BadValue: Badly formatted date: $(AT
0: Exception: BadValue: Badly formatted date: $(AT
4: BadValue: Badly formatted date: $(AT caught in  (/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/develop/sorc/gdas.cd/oops/src/oops/runs/Run.cc +192 execute)
```
This looks to be the same problem you found.   Care needs to be taken as we add more DA capabilities to GDASApp.

I replaced the GDASApp `develop` `yamltools.py` with your `bugfix/socarun` `yamltools.py`.   The 3dvar job was rewound and rebooted.   `fv3jedi_var.x` ran to completion using your `yamltools.py`.

**Review by RussTreadon-NOAA — APPROVED** [review]
The changes to `yamltools.py` in this PR restore the ability to successfully run `fv3jedi_var.x` in the prototype UFS atmospheric DA cycling system.    Without the changes in this PR, `fv3jedi_var.x` aborts as described [here](https://github.com/NOAA-EMC/GDASApp/pull/185#issuecomment-1309679900).

Approve changes to `yamltools.py`

**Comment by RussTreadon-NOAA** [comment]
Also tested updated `yamltools.py` with `fv3jedi_letkf.x` in g-w cycling mode.  It works!

**Comment by CoryMartin-NOAA** [comment]
@jiaruidong2017 let's do those changes in a PR once this is merged in.

**Pull request merged**